pkg/api/pod/util.go

```diff
@@ -774,7 +774,7 @@ func podLifecycleSleepActionZeroValueInUse(podSpec *api.PodSpec) bool {
 			inUse = true
 			return false
 		}
-		if c.Lifecycle.PostStart != nil && c.Lifecycle.PostStart.Sleep != nil && c.Lifecycle.PreStop.Sleep.Seconds == 0 {
+		if c.Lifecycle.PostStart != nil && c.Lifecycle.PostStart.Sleep != nil && c.Lifecycle.PostStart.Sleep.Seconds == 0 {
```

Comment:
How about retooling both of these as:

```
if lc := c.Lifecycle.PostStart; lc != nil && lc.Sleep != nil && lc.Sleep.Seconds == 0 {
```

That way we avoid any such confusion?

Reply:
Oh, I like this idea. Added